### PR TITLE
Data refactor

### PR DESCRIPTION
**Pull request by dirvine**
# Add Pointer Type and Refactor LinkedList Structure

## Changes
* refactor Transacions -> LinkedList (can be single linked or doubly linked) 
* Introduces new `Pointer` type for network addressing
* Makes `LinkedList` outputs optional
* Adds validation and storage handling for network records
* Improves split record handling in networking layer

### New Pointer Type
* Added `Pointer` and `PointerAddress` types for network addressing
* Implemented signature verification and validation for Pointer records
* Added storage and validation handling in `ant-node`
* Added new log markers for tracking Pointer operations
* Introduced `InvalidSignature` error variant for validation failures

### LinkedList Refactoring
* Made `outputs` field optional via `Option<Vec<(PublicKey, LinkedListContent)>>`
* Updated constructors and signature calculation to handle optional outputs
* Modified transaction tests to use `.into()` for empty output vectors
* Maintains backward compatibility with existing functionality

### Networking Layer Updates
* Enhanced split record handling for both Pointer and LinkedList records
* Added proper signature verification for split records
* Implemented count-based selection for handling duplicate records
* Updated record storage validation to handle new types

## Testing
* Added tests for Pointer validation and storage
* Updated transaction tests for optional outputs
* All existing tests pass (excluding known ant-evm issues)

## Impact
These changes improve the flexibility of the network addressing system while making
LinkedList transactions more versatile. The Pointer type provides a new way to handle
network addressing, while optional outputs in LinkedList allow for simpler transaction
structures when needed.

## Breaking Changes
None. All changes maintain backward compatibility with existing functionality.

### CODE REVIEW

File: ant-node/Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 authors = ["MaidSafe Developers <dev@maidsafe.net>"]
 description = "The Autonomi node binary"
 name = "ant-node"
-version = "0.3.1"
+version = "0.3.2"
```
Comment (jacderida) [inline ant-node/Cargo.toml]: I think the version bumping should be left for the release process, so I would revert this.

File: ant-protocol/src/storage/pointer.rs [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    signature: Signature,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub enum PointerTarget {
+    ChunkAddress(ChunkAddress),
+    LinkedListAddress(LinkedListAddress),
+    PointerAddress(PointerAddress),
+    ScratchpadAddress(ScratchpadAddress),
+}
```
Comment (grumbach) [inline ant-protocol/src/storage/pointer.rs:37]: I suppose we omitted registers here on purpose?

**Review by grumbach — APPROVED** [review]
LGTM! 🚢 ✅

File: ant-protocol/src/storage/pointer.rs (hunk 2 above)
Comment (happybeing) [reply ant-protocol/src/storage/pointer.rs:37]: Should this include DataMap, not just for files but arbitrary metadata?

Comment (grumbach) [reply ant-protocol/src/storage/pointer.rs:37]: Agreed +1 @happybeing

**Pull request merged**